src/nodes/node.h

```diff
@@ -312,8 +312,14 @@ class Node : public Serialisable<>
     protected:
     // Persistent data serialisables
     std::map<std::string, std::shared_ptr<SerialisableData>> serialisables_;
+    // Serialise any hidden content
+    virtual void hiddenSerialise(SerialisedValue &target) const {}
+    // Deserialise any hidden content
+    virtual void hiddenDeserialise(const SerialisedValue &node) {}
```

Comment:
I might suggest that "internal" is a better definition than "hidden":

```suggestion
    // Serialise any required internal content
    virtual void serialiseInternal(SerialisedValue &target) const {}
    // Deserialise any required internal content
    virtual void deserialiseInternal(const SerialisedValue &node) {}
```